libr/core/cmd_flag.c

```diff
@@ -867,20 +867,30 @@ static int cmd_flag(void *data, const char *input) {
 				return 0;
 				}	
 			default:
-				addr = r_num_math (core->num, input + 2);
 				break;
 			}
 			core->flags->space_strict = space_strict;
 			f = r_flag_get_at (core->flags, addr, !strict_offset);
 			core->flags->space_strict = false;
-			if (f) {
-				if (f->offset != addr) {
-					r_cons_printf ("%s + %d\n", f->name,
-						(int)(addr - f->offset));
-				} else {
-					r_cons_println (f->name);
-				}
-			}
+            if (f) {
+                if (f->offset != addr) {
+                    // if input contains 'j' print json
+                    if (strchr(input, 'j') != NULL) {
+                        r_cons_printf ("{\"name\":\"%s\",\"offset\":%d}\n",
+                                       f->name, (int)(addr - f->offset));
+                    } else {
+                        r_cons_printf ("%s + %d\n", f->name,
+                                       (int)(addr - f->offset));
+                    }
+                } else {
+                    if (strchr(input, 'j') != NULL) {
+                        r_cons_printf ("{\"name\":\"%s\"}\n",
+                                       f->name);
+                    } else {
+                        r_cons_println (f->name);
+                    }
```

Comment:
use tabs not spaces

Reply:
Sorry, xCode used Indentation default with Space 😬